contracts/CourtRoom.sol

```diff
@@ -7,24 +7,32 @@ import "./ensabstract.sol";
 
 contract CaseContract is Owned {
 
+  struct promise{
+		address beneficiary;
+		uint256 blockNumber;
+		uint8 sig_v;
+		bytes32 sig_r;
+		bytes32 sig_s;
+	}
 	struct claim {
 		bytes32 claimId;
 		address plaintiff;
 		bytes32[] evidence;
+		promise   servicePromise;
 		uint status;
 		bool valid;
```

Comment:
If we assume (as it currently implemented now) that each case/claim is solved upon submission when opening a new claim so we can totally avoid the valid field.
The logs of claims can be kept on the chain by events .
So no need that.